### PR TITLE
Replace Graphik with Fira Sans

### PR DESCRIPTION
**Pull request by daleglass**
Alternate fix for #153

### CODE REVIEW

**Comment by daleglass** [comment]
Notes for testing:

RadioButton.qml contained:
`topPadding: -3 // For perfect alignment when using Graphik`

and Button.qml contained:
`topPadding: -2 // Necessary for proper alignment using Graphik Medium`

So buttons and radio buttons should be looked at to see whether some sort of padding is needed and how much. I commented the line out entirely, so alignment might be wrong with the PR as-is.

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -1096,9 +1096,9 @@ Application::Application(
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Raleway-Bold.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Raleway-SemiBold.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Cairo-SemiBold.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-SemiBold.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-Regular.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-Medium.ttf");
+    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/FiraSans-SemiBold.ttf");
+    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/FiraSans-Regular.ttf");
+    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/FiraSans-Medium.ttf");
```
Comment (ksuprynowicz) [inline interface/src/Application.cpp:1101]: File fonts/FiraSans-Medium.ttf doesn't exist.

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -1096,9 +1096,9 @@ Application::Application(
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Raleway-Bold.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Raleway-SemiBold.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Cairo-SemiBold.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-SemiBold.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-Regular.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-Medium.ttf");
+    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/FiraSans-SemiBold.ttf");
+    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/FiraSans-Regular.ttf");
```
Comment (ksuprynowicz) [inline interface/src/Application.cpp:1100]: This line occurs twice

File: interface/src/Application.cpp [hunk 3]
```diff
@@ -1096,9 +1096,9 @@ Application::Application(
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Raleway-Bold.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Raleway-SemiBold.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Cairo-SemiBold.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-SemiBold.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-Regular.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-Medium.ttf");
+    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/FiraSans-SemiBold.ttf");
```
Comment (ksuprynowicz) [inline interface/src/Application.cpp:1099]: This line occurs twice

**Review by ksuprynowicz — CHANGES_REQUESTED** [review]

File: interface/src/Application.cpp (hunk 1 above)
Comment (daleglass) [reply interface/src/Application.cpp:1099]: done

**Comment by ksuprynowicz** [comment]
This one is ready for testing too :)

**Comment by ksuprynowicz** [comment]
I tested it on Linux and didn't notice any problems.

File: screenshare/src/styles.css [hunk 4]
```diff
@@ -77,8 +77,8 @@ yes:hover + #yes_background {
 }
 
 @font-face {
-  font-family: "Graphik";
-  src: url("./resources/Graphik-Regular.ttf");
+  font-family: "FiraSans";
+  src: url("./resources/FiraSans-Regular.ttf");
```
Comment (JulianGro) [inline screenshare/src/styles.css:81]: According to GitHub, you changed the URL here, but there is no `./resources/FiraSans-Regular.ttf` file

File: tools/jsdoc/hifi-jsdoc-template/static/styles/jsdoc.css [hunk 5]
```diff
@@ -4,12 +4,12 @@
...
 @font-face{
-	font-family: 'Graphik Semibold';
+	font-family: 'FiraSans Semibold';
 	src: url('../fonts/Graphik-SemiBold.otf') format('opentype');
 }
 
 @font-face{
-	font-family: 'Graphik Regular';
+	font-family: 'FiraSans Regular';
 	src: url('../fonts/Graphik-Regular.otf') format('opentype');
```
Comment (JulianGro) [inline tools/jsdoc/hifi-jsdoc-template/static/styles/jsdoc.css:13]: This URLs need to be changed, and the font needs to be put there as well.

File: tools/jsdoc/hifi-jsdoc-template/static/styles/jsdoc.css [hunk 6]
```diff
@@ -4,12 +4,12 @@
 ********************************************************************/
 
 @font-face{
-	font-family: 'Graphik Semibold';
+	font-family: 'FiraSans Semibold';
 	src: url('../fonts/Graphik-SemiBold.otf') format('opentype');
```
Comment (JulianGro) [inline tools/jsdoc/hifi-jsdoc-template/static/styles/jsdoc.css:8]: This URLs need to be changed, and the font needs to be put there as well.

File: launchers/win32/Launcher.rc [hunk 7]
```diff
@@ -77,9 +77,9 @@ IDR_MAINFRAME           ICON                    "res\\interface.ico"
 // BINARY
 //
 
-IDR_FONT_REGULAR        BINARY                  "res\\Graphikregular.ttf"
+IDR_FONT_REGULAR        BINARY                  "res\\FiraSansregular.ttf"
```
Comment (JulianGro) [inline launchers/win32/Launcher.rc:80]: This file does not exist

File: launchers/win32/Launcher.rc [hunk 8]
```diff
@@ -77,9 +77,9 @@ IDR_MAINFRAME           ICON                    "res\\interface.ico"
...
 
-IDR_FONT_BOLD           BINARY                  "res\\Graphikbold.ttf"
+IDR_FONT_BOLD           BINARY                  "res\\FiraSansbold.ttf"
```
Comment (JulianGro) [inline launchers/win32/Launcher.rc:82]: This file does not exist

File: interface/src/Application.cpp [hunk 9]
```diff
@@ -1088,17 +1088,15 @@ Application::Application(
...
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Raleway-Regular.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/rawline-500.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Raleway-Bold.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Raleway-SemiBold.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Cairo-SemiBold.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-SemiBold.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-Regular.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-Medium.ttf");
+    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/FiraSans-SemiBold.ttf");
+    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/FiraSans-Regular.ttf");
```
Comment (JulianGro) [inline interface/src/Application.cpp:1098]: This file does not exist

File: interface/src/Application.cpp [hunk 10]
```diff
@@ -1088,17 +1088,15 @@ Application::Application(
...
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Raleway-Light.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Raleway-Regular.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/rawline-500.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Raleway-Bold.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Raleway-SemiBold.ttf");
     QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Cairo-SemiBold.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-SemiBold.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-Regular.ttf");
-    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/Graphik-Medium.ttf");
+    QFontDatabase::addApplicationFont(PathUtils::resourcesPath() + "fonts/FiraSans-SemiBold.ttf");
```
Comment (JulianGro) [inline interface/src/Application.cpp:1097]: This file does not exist

**Review by JulianGro — CHANGES_REQUESTED** [review]

**Review by JulianGro — CHANGES_REQUESTED** [review]
The interface/resources/qml/stylesUit/GraphikMedium.qml and similar files don't get replaced.
I am pretty sure that this breaks all occurences of HifiStylesUit.FiraSansRegular

File: interface/src/Application.cpp (hunk 9 above)
Comment (daleglass) [reply interface/src/Application.cpp:1098]: It does,
```
interface/resources/fonts/FiraSans-Regular.ttf
interface/resources/fonts/FiraSans-SemiBold.ttf
```

File: interface/src/Application.cpp (hunk 10 above)
Comment (daleglass) [reply interface/src/Application.cpp:1097]: as above

File: launchers/win32/Launcher.rc (hunk 7 above)
Comment (daleglass) [reply launchers/win32/Launcher.rc:80]: We're not using the launcher, but okay, added

**Review by JulianGro — CHANGES_REQUESTED** [review]
There is no replacements for interface/resources/qml/stylesUit/GraphikMedium.qml and similar.

**Comment by daleglass** [comment]
Files added

**Comment by AnimatedMannequin** [comment]
Here are some builds you may use for testing: 

**ubuntu-18.04, full**
 - https://public.overte.org/build/overte/PR155/Overte-PR155-cf90061.sh
 - https://public.overte.org/build/overte/PR155/Overte-PR155-cf90061.tar.gz
 - https://public.overte.org/build/overte/PR155/Overte-PR155-cf90061.tar.Z

**windows-2019, full**
 - https://public.overte.org/build/overte/PR155/Overte-PR155-cf90061.exe

**macOS-10.15, full**
 - https://public.overte.org/build/overte/PR155/Overte-PR155-cf90061.dmg

File: interface/resources/qml/stylesUit/FiraSansRegular.qml [hunk 11]
```diff
@@ -1,14 +1,14 @@
 //
-//  FiraSansRegular.qml
+//  FiraSansMedium.qml
```
Comment (JulianGro) [inline interface/resources/qml/stylesUit/FiraSansRegular.qml:2]:
```suggestion
//  FiraSansRegular.qml
```
Not sure what is going on here

File: interface/resources/qml/stylesUit/FiraSansSemiBold.qml [hunk 12]
```diff
@@ -1,20 +1,21 @@
 //
-//  FiraSansSemiBold.qml
+//  FiraSansMedium.qml
```
Comment (JulianGro) [inline interface/resources/qml/stylesUit/FiraSansSemiBold.qml:2]: Not sure what is going on here
```suggestion
//  FiraSansSemiBold.qml
```

**Review by JulianGro — CHANGES_REQUESTED** [review]

**Comment by AnimatedMannequin** [comment]
Here are some builds you may use for testing: 

**macOS-10.15, full**
 - https://public.overte.org/build/overte/PR155/Overte-PR155-7cd0422.dmg

**ubuntu-18.04, full**
 - https://public.overte.org/build/overte/PR155/Overte-PR155-7cd0422.sh
 - https://public.overte.org/build/overte/PR155/Overte-PR155-7cd0422.tar.Z
 - https://public.overte.org/build/overte/PR155/Overte-PR155-7cd0422.tar.gz

**windows-2019, full**
 - https://public.overte.org/build/overte/PR155/Overte-PR155-7cd0422.exe

**self-hosted_debian-11_aarch64, full**
 - https://public.overte.org/build/overte/PR155/Overte-PR155-7cd0422.tar.gz
 - https://public.overte.org/build/overte/PR155/Overte-PR155-7cd0422.sh
 - https://public.overte.org/build/overte/PR155/Overte-PR155-7cd0422.tar.Z

File: interface/resources/qml/stylesUit/FiraSansSemiBold.qml (hunk 12 above)
Comment (daleglass) [reply interface/resources/qml/stylesUit/FiraSansSemiBold.qml:2]: Copypasta

**Comment by AnimatedMannequin** [comment]
Here are some builds you may use for testing: 

**macOS-10.15, full**
 - https://public.overte.org/build/overte/PR155/Overte-PR155-d214384.dmg

**ubuntu-18.04, full**
 - https://public.overte.org/build/overte/PR155/Overte-PR155-d214384.sh
 - https://public.overte.org/build/overte/PR155/Overte-PR155-d214384.tar.gz
 - https://public.overte.org/build/overte/PR155/Overte-PR155-d214384.tar.Z

**windows-2019, full**
 - https://public.overte.org/build/overte/PR155/Overte-PR155-d214384.exe

**self-hosted_debian-11_aarch64, full**
 - https://public.overte.org/build/overte/PR155/Overte-PR155-d214384.tar.Z
 - https://public.overte.org/build/overte/PR155/Overte-PR155-d214384.sh
 - https://public.overte.org/build/overte/PR155/Overte-PR155-d214384.tar.gz

**Review by JulianGro — APPROVED** [review]

**Comment by JulianGro** [comment]
Just tested JSDOC with the font.
Everything is slightly smaller, but even making it only 5% bigger overshoots it, so I would just leave it as is.

**Comment by JulianGro** [comment]
I don't see anything that needs testing other than JSDOC, since neither screenshare, nor the launcher or the simplified UI are in use or are planned to be used.

**Comment by ksuprynowicz** [comment]
Can we merge this?

**Pull request merged**